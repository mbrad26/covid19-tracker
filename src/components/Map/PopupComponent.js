import React from 'react';
import { Popup } from 'react-map-gl';

import Info from './Info';

const PopupComponent = ({ activeCountry, onSetInactive }) => {
  const handleOnClose = () => onSetInactive();

  return (
    <div id='popup'>
      {activeCountry != null && activeCountry.active &&
        <Popup
          latitude={activeCountry.countryInfo.lat}
          longitude={activeCountry.countryInfo.long}
          onClose={handleOnClose}
          closeOnClick={true}
        >
          <Info activeCountry={activeCountry} />
        </Popup>
      }
    </div>
  )
};

export default PopupComponent;
