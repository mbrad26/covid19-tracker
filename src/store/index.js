import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  applyMiddleware(saga, logger)
);

// console.log('STORE: ', store.getState());

saga.run(rootSaga);

export default store;
