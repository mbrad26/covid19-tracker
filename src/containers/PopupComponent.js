/* istanbul ignore file */
import { connect } from 'react-redux';

import PopupComponent from '../components/Map/PopupComponent';
import { doSetInactive } from '../actions/countryData';
import { getActiveCountry } from '../selectors/data';

const mapStateToProps = ({ countryDataState }) => ({
  activeCountry: getActiveCountry(countryDataState),
});

const mapDispatchToProps = dispatch => ({
  onSetInactive: () => dispatch(doSetInactive()),
});

export default connect(mapStateToProps, mapDispatchToProps)(PopupComponent);
