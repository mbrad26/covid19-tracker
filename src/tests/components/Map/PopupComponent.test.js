import React from 'react';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';

import store from '../../../store/';
import { country } from '../../fixtures';
import PopupComponent from '../../../components/Map/PopupComponent';

describe('PopupComponent', () => {
  const popupProps = {
    activeCountry: null,
    onSetInactive: jest.fn(),
  }

  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <PopupComponent {...popupProps} />
      </Provider>
    );

    screen.debug();

    expect(container.firstChild).toMatchSnapshot();
  });
});
