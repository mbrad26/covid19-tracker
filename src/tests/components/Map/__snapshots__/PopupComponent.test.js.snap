// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupComponent renders snapshot 1`] = `
<div
  id="popup"
/>
`;
